src/librustdoc/html/render/print_item.rs

```diff
@@ -150,7 +150,11 @@ pub(super) fn print_item(cx: &Context<'_>, item: &clean::Item, buf: &mut Buffer,
         clean::StructItem(ref s) => item_struct(buf, cx, item, s),
         clean::UnionItem(ref s) => item_union(buf, cx, item, s),
         clean::EnumItem(ref e) => item_enum(buf, cx, item, e),
-        clean::TypedefItem(ref t, is_associated) => item_typedef(buf, cx, item, t, is_associated),
+        // FIXME: the AssocTypeItem case should never be reached / we should never generate a page
+        //        for an associative type, right? If so, remove the boolean param from item_typedef
+        kind @ (clean::TypedefItem(ref t) | clean::AssocTypeItem(ref t, _)) => {
+            item_typedef(buf, cx, item, t, matches!(kind, clean::AssocTypeItem(..)))
```

Comment:
I was really surprised to see that apparently we generate pages for certain associated types (this branch + the parameter `is_associated` of function `item_typedef` called here)? I don't think that this is reachable for `AssocTypeItem`. Am I wrong? I will gladly simplify `item_typedef` is that's the case.

Reply:
We only generate for typedefs, not for associated typedefs.